File: frontend/apps/ios/Pantopus/Core/Networking/Models/Mailbox/V2/MailboxDocumentDTOs.swift
//
//  MailboxDocumentDTOs.swift
//  Pantopus
//
//  Wire models for `MailboxDocumentEndpoints` — the booklet PDF
//  download and the certified-mail legal delivery proof. Both handlers
//  build their response as a JS object literal, so (unlike the `Mail`
//  row DTOs) the keys are already camelCase.
//

import Foundation

/// `POST /api/mailbox/v2/p2/booklet/:mailId/download` envelope — route
/// `backend/routes/mailboxV2Phase2.js:447`.
public struct BookletDownloadResponse: Decodable, Sendable, Hashable {
    /// Signed URL for the generated PDF.
    public let downloadURL: String?
    /// File size in bytes — RN reports it as MB in the confirmation
    /// (`src/app/mailbox/booklet.tsx:47`).
    public let sizeBytes: Int?

    public init(downloadURL: String?, sizeBytes: Int?) {
        self.downloadURL = downloadURL
        self.sizeBytes = sizeBytes
    }

    private enum CodingKeys: String, CodingKey {
        case downloadURL = "downloadUrl"
        case sizeBytes
    }

    /// "2.4 MB" — the label RN puts in its "Download Started" alert.
    public var megabytesLabel: String? {
        guard let sizeBytes, sizeBytes > 0 else { return nil }
        let megabytes = Double(sizeBytes) / (1024 * 1024)
        return String(format: "%.1f MB", megabytes)
    }
}

/// `GET /api/mailbox/v2/p2/certified/:mailId/proof` envelope — route
/// `backend/routes/mailboxV2Phase2.js:705`.
public struct CertifiedProofResponse: Decodable, Sendable, Hashable {
    public let proof: CertifiedProofDTO?

    public init(proof: CertifiedProofDTO?) {
        self.proof = proof
    }
}

/// The legal delivery proof the backend assembles from the `Mail` row.
public struct CertifiedProofDTO: Decodable, Sendable, Hashable {
    public let mailId: String?
    public let sender: String?
    public let senderTrust: String?
    public let deliveredAt: String?
    public let acknowledgedAt: String?
    public let acknowledgedBy: String?
    public let legalTimestamp: String?

    public init(
        mailId: String?,
        sender: String?,
        senderTrust: String?,
        deliveredAt: String?,
        acknowledgedAt: String?,
        acknowledgedBy: String?,
        legalTimestamp: String?
    ) {
        self.mailId = mailId
        self.sender = sender
        self.senderTrust = senderTrust
        self.deliveredAt = deliveredAt
        self.acknowledgedAt = acknowledgedAt
        self.acknowledgedBy = acknowledgedBy
        self.legalTimestamp = legalTimestamp
    }
}

File: frontend/apps/android/app/src/main/java/app/pantopus/android/data/api/models/mailbox/v2/MailboxDocumentDtos.kt
package app.pantopus.android.data.api.models.mailbox.v2

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Wire models for the document-artefact routes on the phase-2 mailbox
 * router — the booklet PDF download and the certified-mail legal
 * delivery proof. Both handlers build their response as a JS object
 * literal, so (unlike the `Mail` row DTOs) the keys are already
 * camelCase.
 */

/**
 * `POST api/mailbox/v2/p2/booklet/:mailId/download` — route
 * `backend/routes/mailboxV2Phase2.js:447`.
 */
@JsonClass(generateAdapter = true)
data class BookletDownloadResponse(
    /** Signed URL for the generated PDF. */
    @Json(name = "downloadUrl") val downloadUrl: String?,
    /**
     * File size in bytes — RN reports it as MB in the confirmation
     * (`src/app/mailbox/booklet.tsx:47`).
     */
    @Json(name = "sizeBytes") val sizeBytes: Long?,
)

/**
 * `GET api/mailbox/v2/p2/certified/:mailId/proof` — route
 * `backend/routes/mailboxV2Phase2.js:705`.
 */
@JsonClass(generateAdapter = true)
data class CertifiedProofResponse(
    val proof: CertifiedProofDto?,
)

/** The legal delivery proof the backend assembles from the `Mail` row. */
@JsonClass(generateAdapter = true)
data class CertifiedProofDto(
    @Json(name = "mailId") val mailId: String?,
    val sender: String?,
    @Json(name = "senderTrust") val senderTrust: String?,
    @Json(name = "deliveredAt") val deliveredAt: String?,
    @Json(name = "acknowledgedAt") val acknowledgedAt: String?,
    @Json(name = "acknowledgedBy") val acknowledgedBy: String?,
    @Json(name = "legalTimestamp") val legalTimestamp: String?,
)
